src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -709,12 +709,12 @@ def _replace_values_with_deploy_params(
                         final_values_mapping_dict[k].append(
                             self._replace_values_with_deploy_params(item, param_name)
                         )
-                    elif isinstance(v, (str, int, bool)) or not v:
+                    elif isinstance(item, (str, int, bool)) or not v:
```

Comment:
This is a fix for another bug - is it meant to be in this PR?

Reply:
Added it in here so it could just be fixed asap. I'll add it to the PR description